lib/autoupdate/agent/feature_ent.go

```diff
@@ -0,0 +1,26 @@
+//go:build webassets_ent
+// +build webassets_ent
+
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package agent
+
+func init() {
+	featureFlag |= flagEnt
```

Comment:
I copied this from https://github.com/gravitational/teleport/pull/47466. Both the `client` and `agent` packages will use the same flag set from `lib/autoupdate` in an upcoming PR. The build flags should be set when compiled via `teleport.e`. See the top of the file:
```
//go:build webassets_ent
// +build webassets_ent
```

@vapopov I'm noticing now that this flag is only used in `webassets_embed_ent.go`, and the `webassets` prefix is confusing in this non-web-asset context. Would it make sense to look into making this flag more generic?

Reply:
yes, I previously mentioned about this one https://github.com/gravitational/teleport/pull/46587#discussion_r1771903777 since we don't have proper build flag for ent

Reply:
Can we use the `modules` package for this instead of making new flags? This package can already tell you if the binary is OSS, Enterprise, FIPS, etc.

Reply:
Maybe we open a separate PR that renames the build flag so that it can be used in a wider context?

Reply:
I've checked the Makefile of enterprise version, I need to add new build tag, because for tctl and tsh it is not used

Reply:
I don't think `modules.GetModules().IsEnterpriseBuild()` will work as-is for `teleport-update`, since it doesn't have a separate `tool/teleport-update` target that runs `modules.SetModules()`.

That said, we actually want the edition of the agent download to come from the cluster via `/webapi/find` before this ships, so I'll just default it to false and add a TODO for now.